### PR TITLE
feature/TECH 633 docker build arguments

### PR DESCRIPTION
**Pull request by koeves**
Parse docker build arguments from project yaml

----
### 📕 [TECH-633](https://vandebron.atlassian.net/browse/TECH-633) Add docker build arguments danielkoves@vandebron.nl 

🏗️ Build [3](https://jenkins.k8s-serv-backend.vdbinfra.nl/job/MPyL%20Pipeline%20-%20Test/job/PR-266/3/display/redirect) ✅ Successful, started by _Daniel Koves_  
🚀 *[cloudfront-service](https://cloudfront-service-266.test.nl/swagger/index.html)*, *example-dagster-user-code*, *job*, *kong-sync*, *[nodeservice](https://nodeservice-266.test.nl/swagger/index.html)*, *[sbtservice](https://sbtservice-266.test.nl/swagger/index.html)*, *sparkJob*  

[TECH-633]: https://vandebron.atlassian.net/browse/TECH-633?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/mpyl/project.py [hunk 1]
```diff
@@ -392,6 +392,26 @@ def from_config(values: dict):
         return Docker(host_name=values["hostName"])
 
 
+@dataclass(frozen=True)
+class BuildArgs:
+    plain: list[KeyValueProperty]
```
Comment (SamTheisens) [inline src/mpyl/project.py:397]: I'm wondering why this is nested under args. Is there any other flavor than plain?

File: src/mpyl/steps/build/docker_build.py [hunk 2]
```diff
@@ -71,22 +72,32 @@ def execute(self, step_input: Input) -> Output:
             # log in to registry, because we may need to pull in a base image
             login(logger=self._logger, registry_config=docker_registry_config)
 
+        with open(".dockerignore", "w+", encoding="utf-8") as ignore_file:
+            contents = "\n".join(DOCKER_IGNORE_DEFAULT)
+            ignore_file.write(contents)
+
+        build_args: Optional[dict[str, str]] = (
```
Comment (SamTheisens) [inline src/mpyl/steps/build/docker_build.py:79]: Perhaps an empty dict instead of None?

File: src/mpyl/utilities/docker/__init__.py [hunk 3]
```diff
@@ -90,6 +90,7 @@ class DockerConfig:
     build_target: Optional[str]
     test_target: Optional[str]
     docker_file_name: str
+    build_env_file_name: str
```
Comment (SamTheisens) [inline src/mpyl/utilities/docker/__init__.py:93]: This doesn't seem to be used anywhere.

File: src/mpyl/steps/build/docker_build.py (hunk 2 above)
Comment (koeves) [reply src/mpyl/steps/build/docker_build.py:79]: Empty dict is mutable..

Comment (koeves) [reply src/mpyl/steps/build/docker_build.py:79]: python is weird https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments

Comment (koeves) [reply src/mpyl/steps/build/docker_build.py:79]: So the docker.build method should take an optional dict. I guess I can create here an empty dict and map it to None if it's empty.. But it's easier like this I think

**Review by SamTheisens — APPROVED** [review]

**Pull request merged**